### PR TITLE
[UDP Proxy] Add CONNECTION_ID command operator and StreamIdleTimeout response flag

### PR DESCRIPTION
**Pull request by IssaAbuKalbein**
Commit Message:
Additional Description: Add CONNECTION_ID command operator for UDP sessions and set StreamIdleTimeout response flag on session Idle Timeout
Risk Level: low
Testing: unit test, integration tests
Docs Changes:
Release Notes:
Platform Specific Features:

### CODE REVIEW

**Comment by IssaAbuKalbein** [comment]
@danzh2010 Can you, please, take a look on this?

File: test/extensions/filters/udp/udp_proxy/udp_proxy_filter_test.cc [hunk 1]
```diff
@@ -486,15 +487,31 @@ stat_prefix: foo
   test_sessions_[0].recvDataFromUpstream("world3");
   checkTransferStats(17 /*rx_bytes*/, 3 /*rx_datagrams*/, 17 /*tx_bytes*/, 3 /*tx_datagrams*/);
 
+  test_sessions_[0].idle_timer_->invokeCallback();
```
Comment (danzh2010) [inline test/extensions/filters/udp/udp_proxy/udp_proxy_filter_test.cc:490]: Is this used in the test?

File: test/integration/udp_tunneling_integration_test.cc [hunk 2]
```diff
@@ -583,6 +606,8 @@ TEST_P(UdpTunnelingIntegrationTest, IdleTimeout) {
   test_server_->waitForCounterEq("udp.foo.idle_timeout", 1);
   ASSERT_TRUE(upstream_request_->waitForReset());
   test_server_->waitForGaugeEq("udp.foo.downstream_sess_active", 0);
+
+  EXPECT_THAT(waitForAccessLog(access_log_filename), testing::HasSubstr("SI"));
```
Comment (danzh2010) [inline test/integration/udp_tunneling_integration_test.cc:610]: What does "SI" stands for?

File: test/extensions/filters/udp/udp_proxy/udp_proxy_filter_test.cc (hunk 1 above)
Comment (IssaAbuKalbein) [reply test/extensions/filters/udp/udp_proxy/udp_proxy_filter_test.cc]: Removed.

**Comment by IssaAbuKalbein** [comment]
/assign @yanavlasov
@yanavlasov Can you, please, take a look?

**Comment by IssaAbuKalbein** [comment]
@yanavlasov Can you, please, take a look?

**Comment by IssaAbuKalbein** [comment]
@yanavlasov ping

**Comment by adisuissa** [comment]
Assigning @mattklein123 as codeowner
/assign @mattklein123

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Can you add a release note for this change?

/wait

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**